src/timestamp.rs

```diff
@@ -124,4 +142,26 @@ impl Interval {
     pub fn lerp(self, value: f32) -> Timestamp {
         Timestamp((value * (self.duration_ns() as f32)).round() as i64 + self.start.0)
     }
+
+    // convert a string like "500.0 s" to a timestamp
+    pub fn convert_str_to_timestamp(s: &str) -> Result<Timestamp, IntervalParseError> {
+        let mut parts = s.split_whitespace();
+        let value = parts.next().ok_or(IntervalParseError::NoValue)?;
+        let value = value
+            .parse::<f32>()
```

Comment:
Since it shouldn't cost us anything, I wonder if we should parse this as `f64`. It's unlikely, but theoretically possible that we could lose precision with `f32`. (Timestamps are 64-bit for a reason.)